designer/server/src/main/scala/pl/touk/nussknacker/ui/process/periodic/PeriodicProcessesManager.scala

```diff
@@ -0,0 +1,103 @@
+package pl.touk.nussknacker.ui.process.periodic
+
+import com.typesafe.config.Config
+import pl.touk.nussknacker.engine.api.ProcessVersion
+import pl.touk.nussknacker.engine.api.deployment.ProcessActionId
+import pl.touk.nussknacker.engine.api.deployment.periodic.model.DeploymentWithRuntimeParams
+import pl.touk.nussknacker.engine.api.process.{ProcessName, VersionId}
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+import pl.touk.nussknacker.ui.process.periodic.model.PeriodicProcessDeploymentStatus.PeriodicProcessDeploymentStatus
+import pl.touk.nussknacker.ui.process.periodic.model._
+
+import java.time.LocalDateTime
+import scala.concurrent.Future
+
+trait PeriodicProcessesManagerProvider {
```

Comment:
Do we need to keep it abstract? Can't we have one implementation? Also, I feel  confused about responsibility of this class. We have repository, we have service, and this manager what exactly does? It looks like it only forward evething to repository. Why we need to have a provider instead of creating the implementing class directly

Reply:
Removed interfaces PeriodicProcessesManagerProvider, PeriodicProcessesManager. The repository is used directly, without this intermediate layer.